src/notification.c

```diff
@@ -199,17 +199,29 @@ void notification_free(notification * n)
 }
 
 /*
- * Replace all occurrences of "needle" with a quoted "replacement",
- * according to the markup settings.
+ * Replace the two chars where **needle points
+ * with a quoted "replacement", according to the markup settings.
+ *
+ * The needle is a double pointer and gets updated upon return
+ * to point to the first char, which occurs after replacement.
+ *
  */
-char *notification_replace_format(const char *needle, const char *replacement,
-                                  char *haystack, enum markup_mode markup_mode) {
-        char* tmp;
+char *notification_replace_single_field(char *haystack, char **needle,
```

Comment:
If we going to use a double pointer for the needle, I think it'd be better to also use it for the haystack and avoid the awkward assignments in `notification_init`.

Reply:
Is this more "C-ish" than the current signature? I changed it, now for evaluation.